### ISSUE TITLE
[Android] Not working with adjustPan or adjustNothing

### ISSUE THREAD

**Issue by MarioUnlam** [mateusz1913/react-native-avoid-softinput#112]
### Environment

Library version: 3.0.0
OS version: Android Emulator running API 28 (Android 9 - Pie)
React Native: 0.66.3

Dependencies: https://pastebin.com/gwFx5z0k

### Affected platforms

- [X] Android
- [ ] iOS

### Current behavior

Currently, the plugin does not work as I expect. I don't know if I'm missing some config or dependency, or I'm misunderstanding the purpose of this plugin.

If I don't set any **windowSoftInputMode** in AndroidManifest, the app uses the default value (**adjustPan**).
I added this to my screen:

```
React.useEffect(() => {
        AvoidSoftInput.setShouldMimicIOSBehavior(true);
        AvoidSoftInput.setEnabled(true);
    }, []);
```

But the app still uses "**adjustPan**". Running **AvoidSoftInput.setAdjustNothing**() does change the keyboard behavior to "**adjustNothing**", but if the softInput behavior is set to **adjustPan** or **adjustNothing**, the rest of the plugin doesn't work. Opening and closing the keyboard doesn't affect the UI at all. It always uses the default Android behavior for the specified windowSoftInputMode.

I also noticed that adding this to the code (when using **adjustPan** or **adjustNothing**):

```
useSoftInputHeightChanged(({ softInputHeight }) => {
        console.log('softInputHeight', softInputHeight);
    });
```

Only prints "softInputHeight 0" once and nothing else. Opening the keyboard doesn't make it print anything. Seems like the plugin is unable to detect the keyboard when using **adjustPan** or **adjustNothing**.

Now if I add **AvoidSoftInput.setAdjustResize**(), useSoftInputHeightChanged updates when the keyboard is opened. However, the plugin behavior is odd. It adds padding to my form Scrollview, and always scrolls up, even if the focused field is visible when the keyboard opens. There's also a very noticeable delay between opening the keyboard and scrolling up. Besides, the documentation explicitly states that "adjustResize" should not be used alongside this plugin, so I don't understand why it only works when I use that option.

### Expected behavior

The screen I'm testing this on is something like this:

```
<SafeAreaView>
	<StatusBar />
	<View>
		<Formik {...formikInitProps}>
			{(formikProps) => (
				<View>
					{/* This view has flexGrow 1, so it pushes the submit button downwards */}
					<ScrollView>
						{/* Fields */}
					</ScrollView>
				</View>
				<View>
					{/* Submit button at the bottom */}
				</View>
			)}
		</Formik>
	</View>
</SafeAreaView>
```

What I want is to allow all the fields to be visible with the keyboard open, without pushing the submit button upwards. So, the keyboard should open above the submit button, but without occluding the scrollview. And if the focused field ended up being occluded by the keyboard, scroll up so it is visible.

### Reproduction

Can't provide a repo right now. My current project is too complex, and creating a new one will probably lead to a different behavior than the one I'm currently having.

**Comment by mateusz1913** [comment]
Hi @MarioUnlam , thx for reporting the issue.

I would really love, if you could manage to do the reproduction repo as at the moment, there's nothing I can do with it.

If you could create the repro with the code that you provided (it looks like you can omit Formik is setup and just focus on placing UI with correct styles) AND "adjustPan" set in manifest, that would definitely help me understand that use case + I would be able to iterate on it and provide some solution.

Let me know, if it's doable for you to provide, thx in advance!

**Comment by MarioUnlam** [comment]
I tried again in a physical device, using "adjustNothing" in AndroidManifest, and without specifying a different behavior with AvoidSoftInput, and it works now. Maybe it was something related to the Android emulator? In any case, sorry for the false positive.

Still, I found a few differences in the Android version (iOS works great):
- The animation has a small delay. It starts after the keyboard is fully open. Not a big deal if all the delays and durations are set to 0.
- The delay and duration settings don't seem to affect the animation. When opening the keyboard, the screen scrolls up with a fixed duration. When closing the keyboard, the screen scrolls down instantly, without easing. Increasing delay or duration for Show or Hide only makes the screen take longer to "react" to the keyboard, but doesn't affect the animation.
- If a scrollview has more content that can be rendered above the keyboard, then when you focus on a field, it will **always** scroll up, even if that field was fully visible when the keyboard opened. This is a problem, because if you focus a field that is close to the top of your screen, it can end up outside the screen, or cut off.

But these are different issues than the one in the original post. I just wanted to mention them here to avoid creating another issue, and I assume some, if not all of them, are already known. I think for now I'll just use this library for iOS.
